src/mesh/tdycoremesh.c

```diff
@@ -2268,14 +2268,92 @@ PetscBool AreCellsNeighbors(TDy tdy, PetscInt cell_id_1, PetscInt cell_id_2) {
 }
 
 /* -------------------------------------------------------------------------- */
-PetscErrorCode ArrangeCellsInAnCircularOrder(TDy tdy, PetscInt *cellsAbvBlw, PetscInt ncells, PetscInt *cell_order) {
+PetscErrorCode ArrangeCellsInCircularOrder(TDy tdy, PetscInt *cellsAbvBlw, PetscInt ncells, PetscInt *cell_order) {
 
   PetscFunctionBegin;
+
+  // 1. If all cells in cellsAbvBlw do not have same number of neighbors, then
+  //    - Check to ensure only two cells have miniumum number of neighbors, and
+  //    - Arrange cellsAbvBlw such that cells with minimum number of neighbors are
+  //      at the start and end of the array.
+
+  PetscInt nNeighbors[ncells];
+  for (PetscInt ii = 0; ii < ncells; ii++) {
+    nNeighbors[ii] = 0;
+  }
+
+  // Determine the number of neigbhors
+  for (PetscInt ii = 0; ii < ncells; ii++){
+    for (PetscInt jj = ii+1; jj < ncells; jj++){
+      nNeighbors[ii] += AreCellsNeighbors(tdy, cellsAbvBlw[ii], cellsAbvBlw[jj]);
+      nNeighbors[jj] += AreCellsNeighbors(tdy, cellsAbvBlw[ii], cellsAbvBlw[jj]);
+    }
+  }
+
+  // Determine (1) min/max number of neighbors, and (2) index of cell
+  // with minimum number of cells
+  PetscInt minIdx = -1, minNeighbors = ncells, maxNeighbors = 0;
+  for (PetscInt ii = 0; ii < ncells; ii++) {
+    if (nNeighbors[ii] < minNeighbors) {
+      minNeighbors = nNeighbors[ii];
+      minIdx = ii;
+    }
+    if (nNeighbors[ii] > maxNeighbors) {
+      maxNeighbors = nNeighbors[ii];
+    }
+  }
+
+  if (minNeighbors != maxNeighbors) {
+
+    // Cells have different number of neighbors
+
+    // Check only two cells have min number of neighbors
+    PetscInt count;
+    count = 0;
+    for (PetscInt ii = 0; ii < ncells; ii++) {
+      if (nNeighbors[ii] == minNeighbors) {
+        count++;
+      }
+    }
+    if (count != 2) {
+      SETERRQ(PETSC_COMM_WORLD,PETSC_ERR_USER,"Number of cells with minimum number of neighbors is not TWO");
+    }
+
+    // Put the cells with min number of neigbhors at the start and end of the array
+    PetscInt tmpOrder[ncells];
+    count = 0;
+    tmpOrder[0] = cellsAbvBlw[minIdx];
+    for (PetscInt ii = 0; ii < ncells; ii++){
+      if (ii != minIdx && nNeighbors[ii] > minNeighbors) {
+        count = count + 1;
```

Comment:
This is trivial, but one of C's idioms (also used in C++) is that `count++` can be used for `count = count + 1`, and `count += n` can be used instead of `count = count + n`. Once you get used to it, it's much more "breezy".